rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -662,3 +843,171 @@ def prediction_states(
         self._choose_last_user_input(trackers_as_states, use_text_for_last_user_input)
 
         return trackers_as_states
+
+
+class IntentMaxHistoryTrackerFeaturizer(MaxHistoryTrackerFeaturizer):
+
+    LABEL_NAME = "intent"
+
+    def _convert_labels_to_ids(
+        self, trackers_as_intents: List[List[Text]], domain: Domain
+    ) -> np.ndarray:
+        """Convert a list of labels to an np.ndarray of label ids.
+        The number of rows is equal to `len(trackers_as_intents)`.
+        The number of columns is equal to the maximum number of labels
+        that any Labels item has. Rows are padded with -1 if not all Labels 
+        items have the same number of labels.
+
+        Returns:
+            A 2d np.ndarray of label ids.
+        """
+
+        # store labels in numpy arrays so that it corresponds to np arrays of input features
+        label_ids = [
+            [
+                domain.intents.index(intent)
+                for intent in tracker_intents
+            ]
+            for tracker_intents in trackers_as_intents
+        ]
+
+        pad_val = -1
+
+        # Add -1 padding to labels array so that
+        # each example has equal number of labels
+        multiple_labels_count = [len(a) for a in label_ids]
+        max_labels_count = max(multiple_labels_count)
+        num_padding_needed = [max_labels_count - len(a) for a in label_ids]
+
+        new_label_ids = []
+        for ids, num_pads in zip(label_ids, num_padding_needed):
+            if num_pads:
+                ids.extend([pad_val] * num_pads)
+            new_label_ids.append(ids)
```

Comment:
Can you package this into a helper method? It would make the code a bit easier to read.

Reply:
Will do!